src/cargo/core/source.rs

```diff
@@ -223,9 +225,39 @@ impl SourceId {
         Ok(SourceId::for_registry(&url))
     }
 
+    pub fn alt_registry(config: &Config, key: &str) -> CargoResult<SourceId> {
+        let registries = config.get_table("registries")?;
+        match registries.as_ref().and_then(|registries| registries.val.get(key)) {
+            Some(registry)  => {
+                let index = match *registry {
+                    CV::String(ref s, _)        => s,
```

Comment:
It looks like this is supporting both:

```toml
[registries]

foo = "https://path/to/index"
```

and

```
[registries.foo]
index = "https://path/to/index"
```

Should we perhaps canonicalize on just one? (or was it a nice-to-have to support both?)

Reply:
I did this because its similar to how dependencies work (with the `version` key), in case we ever allow optional additional fields in registries

Reply:
Indeed yeah, although that was primarily done for ergonomics (as you're basically always doing that). In this case though, maybe we could just canonicalize on the latter to start out with and extend it in the future if necessary?

Reply:
I don't have a strong opinion. :)